x-pack/test/rbac_api_integration/apis/saved_objects/find.js

```diff
@@ -290,7 +246,7 @@ export default function ({ getService }) {
         noType: {
           description: 'all objects',
           statusCode: 200,
-          response: expectAllResultsIncludingInvalidTypes,
```

Comment:
Are we sure that nothing is relying on the legacy behavior of returning invalid types?

Reply:
All of the current usages of the SavedObjectsClient inside of Kibana and x-pack specify a type when doing a `find`, so none of them are relying upon this functionality. I can't think of any valid scenarios where we'd want to do these to be returned as they have to be malformed documents , I mainly added these asserts and the test bad data when trying to think of any edge-cases that we'd be introducing with RBAC Phase 1, but I'm not sure it's really relevant at all.